src/main/java/gyro/google/compute/SnapshotResource.java

```diff
@@ -0,0 +1,433 @@
+        this.storageLocations = storageLocations;
+    }
+
+    /**
+     * The status of snapshot creation. Values can be: ``CREATING``, ``UPLOADING``, ``FAILED``, ``READY``, or ``DELETING``.
+     */
+    @Output
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+
+    /**
+     * The ID value of the disk used to create the snapshot.
+     */
+    @Output
+    public String getSourceDiskId() {
+        return sourceDiskId;
+    }
+
+    public void setSourceDiskId(String sourceDiskId) {
+        this.sourceDiskId = sourceDiskId;
+    }
+
+    /**
+     * The size of the disk used to create the snapshot, specified in GB.
+     */
+    @Output
+    public Long getDiskSizeGb() {
+        return diskSizeGb;
+    }
+
+    public void setDiskSizeGb(Long diskSizeGb) {
+        this.diskSizeGb = diskSizeGb;
+    }
+
+    /**
+     * The size of the storage used by the snapshot. This is expected to change with snapshot creation/deletion as snapshots share storage.
+     */
+    @Output
+    public Long getStorageBytes() {
+        return storageBytes;
+    }
+
+    public void setStorageBytes(Long storageBytes) {
+        this.storageBytes = storageBytes;
+    }
+
+    /**
+     * The fully-qualified URL linking back to the snapshot.
+     */
+    @Output
+    @Id
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink != null ? toSnapshotUrl(getProjectId(), selfLink) : null;
+    }
+
+    /**
+     * The fingerprint for the labels being applied to the snapshot, which is updated any time the labels change.
+     */
+    @Output
+    public String getLabelFingerprint() {
+        return labelFingerprint;
+    }
+
+    public void setLabelFingerprint(String labelFingerprint) {
+        this.labelFingerprint = labelFingerprint;
+    }
+
+    @Override
+    public void copyFrom(Snapshot snapshot) {
+        setName(snapshot.getName());
+        setDescription(snapshot.getDescription());
+        setLabels(snapshot.getLabels());
+        setStorageLocations(snapshot.getStorageLocations());
+        setStatus(snapshot.getStatus());
+        setSourceDiskId(snapshot.getSourceDiskId());
+        setDiskSizeGb(snapshot.getDiskSizeGb());
+        setStorageBytes(snapshot.getStorageBytes());
+        setSelfLink(snapshot.getSelfLink());
+        setLabelFingerprint(snapshot.getLabelFingerprint());
+
+        if (DiskResource.parseDisk(getProjectId(), snapshot.getSourceDisk()) != null) {
+            setSourceDisk(findById(DiskResource.class, snapshot.getSourceDisk()));
+        }
+
+        if (RegionDiskResource.parseRegionDisk(getProjectId(), snapshot.getSourceDisk()) != null) {
+            setSourceRegionDisk(findById(RegionDiskResource.class, snapshot.getSourceDisk()));
+        }
+
+        if (snapshot.getSourceDiskEncryptionKey() != null) {
```

Comment:
setSourceDiskEncryptionKey(null);
```suggestion
        if (snapshot.getSourceDiskEncryptionKey() != null) {
```